### ISSUE TITLE
Declared variables are not accessible to the memory storage in C# until a value is set

### ISSUE THREAD

**Issue by McJones** [YarnSpinnerTool/YarnSpinner-Unity#100]
**What is the current behavior?**

Any variables declared in the yarn script itself have no value that the C# can access until its value is set once in the yarn.
Once the variable has a value set it is then accessible via C#, but will always fail to be found before then.

**Please provide the steps to reproduce, and if possible a minimal demo of the problem**:

1. Create a new default Yarn Spinner project
2. Declare some Yarn values in the script
3. Attempt to access those values via C#

Example yarn file
```
title: Start
---
<<declare $num to 1>>
===
```

Example C# file:
```csharp
var storage = GameObject.FindObjectOfType<Yarn.Unity.InMemoryVariableStorage>();
int num;
if (!storage.TryGetValue("$num", out num))
{
    Debug.Log("it doesn't exist");
    return;
}
```

**What is the expected behavior?**

The declared variables will be accessible via C# without needing a value to be set, otherwise why have defaults?
So in the above example files the C# `int num` would have the value `1` which it got from the `$num` variable via Yarn.

**Please tell us about your environment:**
  
  - Yarn Spinner Version: v2.0.0-beta4
  - Unity Version: 2019.4.11f1

**Other information**

**Comment by McJones** [comment]
As a workaround for now you can extract the values from the Yarn Program and inject them back into the memory storage:

```csharp
var project = GameObject.FindObjectOfType<Yarn.Unity.DialogueRunner>().yarnProject;
var values = project.GetProgram().InitialValues;
foreach (var pair in values)
{
    var value = pair.Value;
    switch (value.ValueCase)
    {
        case Yarn.Operand.ValueOneofCase.StringValue:
            storage.SetValue(pair.Key, value.StringValue);
            break;
        case Yarn.Operand.ValueOneofCase.BoolValue:
            storage.SetValue(pair.Key, value.BoolValue);
            break;
        case Yarn.Operand.ValueOneofCase.FloatValue:
            storage.SetValue(pair.Key, value.FloatValue);
            break;
        default:
            Debug.LogWarning($"{pair.Key} is an invalid type");
            break;
    }
}
```

**Comment by McJones** [comment]
Closed in commit ac21122
